lib/msf/core/payload/windows/encrypted_reverse_tcp.rb

```diff
@@ -20,6 +20,25 @@ module Payload::Windows::EncryptedReverseTcp
 
   def initialize(*args)
     super
+
+    # prevents checks running when module is initialized during msfconsole startup
+    if framework
+      unless framework.db.connection_established?
+        add_error("This module requires a database connection. Please run 'msfdb init' or 'msfdb start'.")
```

Comment:
A DB connection is preferred, but not necessarily a requirement since the `ChachaKey` and `ChachaNonce` options can be used instead. Maybe replace this with a warning letting the user know to use those options instead?

Reply:
Good shout! Will update the message and reduce the method from an `add_error` to an `add_warning`